<launch>
    <arg name="topic_camera_info_sub"         default="/sick_visionary_t_mini/camera_info"/>
    <arg name="topic_color_image_sub"         default="/sick_visionary_t_mini/intensity"/>
    <arg name="topic_depth_image_sub"         default="/sick_visionary_t_mini/points"/>  
    <arg name="distance_field_detect"         value="1.7"/> 
    <arg name="distance_field_warning"        value="1.2"/> 
    <arg name="distance_field_dangerous"      value="0.7"/>

    <arg name="camera"                        default=""/>
    <arg name="usb_port_id"                   default=""/>
    <arg name="width_image_resize"            value="500"/>
    <arg name="height_image_resize"           value="300"/>


    <include file="$(find sick_visionary)/launch/sick_visionary-t_mini.launch"></include>

    <node name="detect_obstacle" pkg="convert_image" type="detect_obstacle_pointcloud2.py" output="screen">
        <param name="use_detect"                        value="true" />
        <param name="use_rotate"                        value="true" />
        <param name="is_display_origin_color_image"     value="false" />
        <param name="is_display_origin_depth_image"     value="false" />
        <param name="is_display_resize_depth_image"     value="true" />
        <param name="camera"                            value="$(arg camera)" />
        <param name="topic_camera_info_sub"             value="$(arg topic_camera_info_sub)" />
        <param name="topic_color_image_sub"             value="$(arg topic_color_image_sub)" />
        <param name="topic_depth_image_sub"             value="$(arg topic_depth_image_sub)" />
        <param name="width_image_resize"                value="$(arg width_image_resize)" />
        <param name="height_image_resize"               value="$(arg height_image_resize)" />
        <param name="distance_field_detect"             value="$(arg distance_field_detect)" />
        <param name="distance_field_warning"            value="$(arg distance_field_warning)" />
        <param name="distance_field_dangerous"          value="$(arg distance_field_dangerous)" />
    </node>
    <!-- <node name="view" pkg="convert_image" type="view.py" output="screen"></node> -->


    <arg name="open_rviz" default="false"/>
    <group if="$(arg open_rviz)"> 
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find convert_image)/rviz/sick_visionary.rviz" required="true" />
    </group>
</launch>